deploy/chart/templates/persistencevolumeclaim.yaml

```diff
@@ -0,0 +1,39 @@
+{{- if and .Values.mosquitto.persistence.enabled -}}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ include "mosquitto.fullname" . }}
+  labels: {{- include "mosquitto.labels.standard" . | nindent 4 }}
+    {{- with .Values.mosquitto.common.labels }}
+    {{- toYaml . | trim | nindent 4 }}
+    {{- end }}
+    {{- with .Values.mosquitto.persistentVolumeClaim.labels }}
+    {{- toYaml . | trim | nindent 4 }}
+    {{- end }}
+  {{- if or .Values.mosquitto.common.annotations .Values.mosquitto.persistentVolumeClaim.annotations }}
+  annotations:
+    {{- with .Values.mosquitto.common.annotations }}
+    {{- toYaml . | trim | nindent 4 }}
+    {{- end }}
+    {{- with .Values.mosquitto.persistentVolumeClaim.annotations }}
+    {{- toYaml . | trim | nindent 4 }}
+    {{- end }}
+  {{- end }}
+spec:
+  {{- with .Values.mosquitto.persistentVolumeClaim.accessModes }}
+  accessModes:
+    {{- range . }}
+    - {{ . }}
+    {{- end }}
+  {{- end }}
+  resources:
+    requests:
+      storage: {{ required ".Values.mosquitto.persistence.size is required!" .Values.mosquitto.persistence.size }}
+  {{- with .Values.mosquitto.persistentVolumeClaim.storageClassName }}
+  {{- if eq "-" . }}
+  storageClassName: ""
+  {{- else }}
+  storageClassName: {{ . }}
+  {{- end }}

```

Comment:
could you add here the properties to managed static provisioned volumes? This enables to keep the volumes on the NAS/SAN on cluster reinstall.

```
{{- if .Values.persistence.existingVolume }}
      volumeName: {{ .Values.persistence.existingVolume }}
      {{- end }}
      {{- if or .Values.persistence.matchLabels (.Values.persistence.matchExpressions) }}
      selector:
      {{- if .Values.persistence.matchLabels }}
        matchLabels:
        {{ toYaml .Values.persistence.matchLabels | indent 8 }}
      {{- end -}}
      {{- if .Values.persistence.matchExpressions }}
        matchExpressions:
          {{ toYaml .Values.persistence.matchExpressions | indent 8 }}
        {{- end -}}
      {{- end }}
```
Here you have a full sample if you want: https://github.com/pajikos/home-assistant-helm-chart/pull/48/files

Reply:
I have added this enhancement. It would be great if you could review it

Reply:
it LGTM, in any case as soon as the chart goes out, I'll try it out, so I can do some fixes if required,